recipes-kernel/kernel-tests/kernel-performance-tests-files/upload_cyclictest_results.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python3
+"""Script for uploading a cyclictest log to influxdb"""
+
+import argparse
+import logging
+import mmap
+import re
+import subprocess
+import sys
+from pathlib import Path
+from datetime import datetime
+from influxdb import InfluxDBClient, exceptions
+import requests.exceptions
+
+
+def get_current_kernel_version():
+    """Get the kernel version currently running on the system"""
+    try:
+        process = subprocess.run(['uname', '-r'],
+                                 check=True,
+                                 stdout=subprocess.PIPE)
+        kver_full = process.stdout.strip().decode("utf-8")
+        rgx = re.search(r'([0-9]+\.[0-9]+)\.([0-9]+)', kver_full)
+        kver = rgx.group(1)
+    except (FileNotFoundError, AttributeError):
+        logging.warning("failed to read the current kernel version")
+        raise
+
+    return kver, kver_full
+
+
+def get_kernel_version(path):
+    """Get the kernel version used in cyclictest test
+
+    Read 'major.minor' and 'major.minor.patch-rt' kernel version strings from
+    cyclictest log if found or current running system otherwise.
+
+    """
+    try:
+        with open(path, 'rb', 0) as file, \
+             mmap.mmap(file.fileno(), 0, access=mmap.ACCESS_READ) as mfile:
```

Comment:
Nice. Did you see a performance improvement in doing this versus reading the file line by line and parsing?

Reply:
I have not done any benchmarking yet - saw it as a recommended way to do this kind of regex searching through a file and it seems to work pretty well.